tests/dryrun/util.go

```diff
@@ -0,0 +1,173 @@
+
+func embedReleaseData() error {
+	if err := release.SetReleaseDataForTests(map[string][]byte{
+		"release.yaml": []byte(releaseData),
+	}); err != nil {
+		return fmt.Errorf("set release data: %v", err)
+	}
+	return nil
+}
+
+func runEmbeddedClusterCmd(args ...string) error {
+	fullArgs := append([]string{"embedded-cluster"}, args...)
+	os.Args = fullArgs // for reporting
+	return cmd.NewApp("embedded-cluster").Run(fullArgs)
+}
+
+func readK0sConfig(t *testing.T) k0sv1beta1.ClusterConfig {
+	stdout, err := exec.Command("cat", defaults.PathToK0sConfig()).Output()
+	if err != nil {
+		t.Fatalf("fail to get k0s config: %v", err)
+	}
+	k0sConfig := k0sv1beta1.ClusterConfig{}
+	if err := yaml.Unmarshal(stdout, &k0sConfig); err != nil {
+		t.Fatalf("fail to unmarshal k0s config: %v", err)
+	}
+	return k0sConfig
+}
+
+func assertCollectors(t *testing.T, actual []*troubleshootv1beta2.HostCollect, expected map[string]struct {
+	match    func(*troubleshootv1beta2.HostCollect) bool
+	validate func(*troubleshootv1beta2.HostCollect)
+}) {
+	found := make(map[string]bool)
+	for _, collector := range actual {
+		for name, assertion := range expected {
+			if assertion.match(collector) {
+				found[name] = true
+				assertion.validate(collector)
+			}
+		}
+	}
+	for name := range expected {
+		assert.True(t, found[name], fmt.Sprintf("%s collector not found", name))
+	}
+}
+
+func assertAnalyzers(t *testing.T, actual []*troubleshootv1beta2.HostAnalyze, expected map[string]struct {
+	match    func(*troubleshootv1beta2.HostAnalyze) bool
+	validate func(*troubleshootv1beta2.HostAnalyze)
+}) {
+	found := make(map[string]bool)
+	for _, collector := range actual {
+		for name, assertion := range expected {
+			if assertion.match(collector) {
+				found[name] = true
+				assertion.validate(collector)
+			}
+		}
+	}
+	for name := range expected {
+		assert.True(t, found[name], fmt.Sprintf("%s collector not found", name))
+	}
+}
+
+func assertMetrics(t *testing.T, actual []dryruntypes.Metric, expected map[string]func(string)) {
+	found := make(map[string]bool)
+	for _, m := range actual {
+		if validator, ok := expected[m.Title]; ok {
+			found[m.Title] = true
+			validator(m.Payload)
+		}
+	}
+	for title := range expected {
+		assert.True(t, found[title], fmt.Sprintf("%s metric not found", title))
+	}
+}
+
+func assertEnv(t *testing.T, actual, expected map[string]string) {
+	for expectedKey, expectedValue := range expected {
+		assert.Equal(t, expectedValue, actual[expectedKey])
+	}
+}
+
+func assertConfigMapExists(t *testing.T, kcli client.Client, name string, namespace string) {
+	var cm corev1.ConfigMap
+	err := kcli.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, &cm)
+	assert.NoError(t, err)
+}
+
+func assertSecretExists(t *testing.T, kcli client.Client, name string, namespace string) {
+	var secret corev1.Secret
+	err := kcli.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, &secret)
+	assert.NoError(t, err)
```

Comment:
should you check the error here for not found?

Reply:
why would i only check "not found" error?